overlord/devicestate/handlers_install.go

```diff
@@ -1214,14 +1252,102 @@ func (m *DeviceManager) doInstallFinish(t *state.Task, _ *tomb.Tomb) error {
 	if err := t.Get("on-volumes", &onVolumes); err != nil {
 		return err
 	}
-	logger.Debugf("install-finish for %q on %v", systemLabel, onVolumes)
-	// TODO: use the seed to get gadget/kernel
-	// - install missing volumes structure content
-	// - install gadget assets
-	// - install kenrel
-	// - make system bootable (include writing modeenv)
+	logger.Debugf("starting install-finish for %q systemd on %v", systemLabel, onVolumes)
+
+	essentialTypes := []snap.Type{snap.TypeKernel, snap.TypeBase, snap.TypeGadget}
+	st.Unlock()
+	sys, snapInfos, snapSeeds, err := m.loadSystemAndEssentialSnaps(systemLabel, essentialTypes)
+	st.Lock()
+	if err != nil {
+		return err
+	}
+	// Unset revision means the snap will be local, assign a local
+	// revision so we can seed/install the snap.
```

Comment:
the way you tweaked my comment it doesn't have the right meaning,  we are not going to install or seed the snap here. I really meant we are doing the equivalent thing of, not a prereq for:

``
// Unset revision means the snap will be local, assign a local
// revision as seeding/installing the snap does
``

Reply:
Changed